pkg/didcomm/protocol/transport/transport_interface.go

```diff
@@ -0,0 +1,30 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package transport
+
+// OutboundTransport interface definition for transport layer
+// This is the client side of the agent
+type OutboundTransport interface {
+	// Send send a2a exchange data
+	Send(data string, destination string) (string, error)
+}
+
+// RequestRouter struct for path and handler function
+type RequestRouter struct {
+	Path        string
+	HandlerFunc func([]byte) error
```

Comment:
I'm uncertain about this pattern in this context.

If you create function signatures, they should be created as a type with a comment. e.g., https://golang.org/src/net/http/server.go?s=60931:60978#L1981

Reply:
actually, we can use `http.HandlerFunc` as type. Anyway, inbound handling will be done in a separate PR